File: 73-set-matrix-zeroes/set-matrix-zeroes.cpp
class Solution {
public:
    void setZeroes(vector<vector<int>>& matrix) {
        int row = matrix.size();
        int col = matrix[0].size();

        vector<vector<int>> directions = {{1, 0}, {0, 1}, {-1, 0}, {0, -1}};

        unordered_set<int> zeroR;
        unordered_set<int> zeroC;

        for (int i = 0; i < row; i++) {
            for (int j = 0; j < col; j++) {
                if (matrix[i][j] == 0) {
                    zeroR.insert(i);
                    zeroC.insert(j);
                }
            }
        }

        for (int r : zeroR) {
            for (int j = 0; j < col; j++) {
                matrix[r][j] = 0;
            }
        }

        for (int c : zeroC) {
            for (int j = 0; j < row; j++) {
                matrix[j][c] = 0;
            }
        }
    }
};

File: 73-set-matrix-zeroes/set-matrix-zeroes.py
class Solution:
    def setZeroes(self, matrix: List[List[int]]) -> None:
        row = len(matrix)
        col = len(matrix[0])

        zeroRows = set()
        zeroCols = set()

        for i in range(row):
            for j in range(col):
                if matrix[i][j] == 0:
                    zeroRows.add(i)
                    zeroCols.add(j)

        for r in zeroRows:
            for j in range(col):
                matrix[r][j] = 0
        
        for c in zeroCols:
            for i in range(row):
                matrix[i][c] = 0